### ISSUE TITLE
Navigation errors

### ISSUE THREAD

**Issue by rickardinho** [DRDD2016/native#251]
This might have something to do with some of the other bugs raised.  The more i use the app the more that i seem to get the wrong screen show when tapping on tabbar icons.  
Example I tap the Feed I get Code screen.  I tap Create event I get Confirmed event screen.  I tap Code and get a Poll screen.  It Seems to want to to go back to a screen I've been on one or two screens before rather than redirect to the correct screen corresponding to the tab.   Do you get the same behaviour?

**Comment by rickardinho** [comment]
Some screenshots which demonstrate this.  Note tabbar shows Feed highlighted, but screens are not the Feed:

![img_5444](https://cloud.githubusercontent.com/assets/17908724/23531868/7878bc06-ffa0-11e6-8d93-dab9b88ddad9.PNG)
![img_5445](https://cloud.githubusercontent.com/assets/17908724/23531875/7f158602-ffa0-11e6-89b2-bbcbc6bffcf2.PNG)

**Comment by minaorangina** [comment]
@rickardinho This warrants a proper phone conversation, but just to give you a quick rundown of where things are here (I'll repeat stuff you know already but bear with me):

There are some fundamental differences between how navigation works in mobile development and web development - we touched upon this last Friday.  Essentially, each tab navigates to a stack of views, as opposed to a specific page.  This is unlike the web, where a link corresponds to a specific page.  So when moving from tab to tab, you're moving from stack to stack.   

When viewing an event, it's considered the next view on the stack, be it the code, calendar or feed stack.  When moving between tabs, the fact you can still see an event page, instead of the feed or calendar, is actually a deliberate feature of mobile navigation, so that the phone can remember where you left off in that particular stack.  

In short, nothing is actually broken, just that the concept of the app was originally designed with web navigation in mind, as opposed to mobile navigation.  We can't force the app to behave as a web app would, so really the long-term solution would be some sort of redesign.

If a user wants to 'go back to the beginning', the usual way is to double-tap the tab to 'reset the stack'.  The instagram app is a good example of this.  We've implemented the double-tap-to-reset in a recent PR (doesn't seem to be a way to reset the stack with only one tap - we tried).

Again, we can discuss in more depth over the phone later this week.

**Comment by rickardinho** [comment]
I don't think the user cares whether they are on a 'stack' or not.  However, I don't think the issue is as you describe above.  This wouldn't explain why I get the Code screen when clicking on the Feed would it?

Can't we just say every screen is a new stack?

This needs to be fixed as its not fit for purpose if the user clicks on a tab (i.e. Feed) and doesn't get the correct screen (i.e. Feed).

**Comment by minaorangina** [comment]
We're on the same page. As I say, we will discuss in more depth over the phone later this week. There's nothing we can change about how mobile navigation works, but we'll look to see if there's any workaround

**Comment by minaorangina** [comment]
Also do you have an issue for getting seeing code when you click on feed?  I haven't seen it

**Comment by rickardinho** [comment]
Yes, see the first comment in this thread, and the second screenshot.  I clicked Feed and go the Code screen (you can see Feed is highlighted).

**Comment by heron2014** [comment]
@rickardinho As per our conversation over the phone, we have enhanced the navigation by moving the event view into its own navigation stack as a modal style screen.  Please test.

**Comment by rickardinho** [comment]
Looks to work pretty well to me.  Closing
